sklearn/tree/_utils.pxd

```diff
@@ -104,3 +105,49 @@ cdef class PriorityHeap:
                   double impurity, double impurity_left,
                   double impurity_right) nogil
     cdef int pop(self, PriorityHeapRecord* res) nogil
+
+# =============================================================================
+# WeightedPQueue data structure
```

Comment:
`P` is shorthand for "priority", since it's a queue ordered by the `data` attribute of the samples. Is there a more clear / appropriate name you think i should use?